File: aladdinworks3_ui/src/services/VendorService.js
import http from "../http-common"; 

class VendorService {
  getAllVendors(searchDTO) {
    console.log(searchDTO)
    return this.getRequest(`/vendor/vendors`, searchDTO);
  }

  get(vendorId) {
    return this.getRequest(`/vendor/${vendorId}`, null);
  }

  findByField(matchData) {
    return this.getRequest(`/vendor?field=${matchData}`, null);
  }

  addVendor(data) {
    return http.post("/vendor/addVendor", data);
  }

  update(data) {
  	return http.post("/vendor/updateVendor", data);
  }
  
  uploadImage(data,vendorId) {
  	return http.postForm("/vendor/uploadImage/"+vendorId, data);
  }




	postRequest(url, data) {
		return http.post(url, data);
      };

	getRequest(url, params) {
        return http.get(url, {
        	params: params
        });
    };

}

export default new VendorService();

File: src/main/java/com/aladdinworks3/service/VendorService.java
package com.aladdinworks3.service;

import java.util.List;

import org.springframework.data.domain.Page;
import org.springframework.data.domain.Pageable;
import org.springframework.data.jpa.domain.Specification;
import org.springframework.http.ResponseEntity;

import com.aladdinworks3.domain.Vendor;
import com.aladdinworks3.dto.VendorDTO;
import com.aladdinworks3.dto.VendorSearchDTO;
import com.aladdinworks3.dto.VendorPageDTO;
import com.aladdinworks3.dto.VendorConvertCriteriaDTO;
import com.aladdinworks3.service.GenericService;
import com.aladdinworks3.dto.common.RequestDTO;
import com.aladdinworks3.dto.common.ResultDTO;
import java.util.List;
import java.util.Optional;





public interface VendorService extends GenericService<Vendor, Integer> {

	List<Vendor> findAll();

	ResultDTO addVendor(VendorDTO vendorDTO, RequestDTO requestDTO);

	ResultDTO updateVendor(VendorDTO vendorDTO, RequestDTO requestDTO);

    Page<Vendor> getAllVendors(Pageable pageable);

    Page<Vendor> getAllVendors(Specification<Vendor> spec, Pageable pageable);

	ResponseEntity<VendorPageDTO> getVendors(VendorSearchDTO vendorSearchDTO);
	
	List<VendorDTO> convertVendorsToVendorDTOs(List<Vendor> vendors, VendorConvertCriteriaDTO convertCriteria);

	VendorDTO getVendorDTOById(Integer vendorId);







}





